action.yaml

```
@@ -394,16 +394,16 @@ runs:
           }
 
           # skip if already there
-          if (pkg_ref$ref %in% d$get_field("Config/Needs/DepsDev")) {
+          if (pkg_ref$ref %in% d$get_field("Config/Needs/DepsDev", ""))) {
```

Hmm this is even more buggy :(
`d$get_field` returns you a big string with values separated by commas so in order to check if `pkg_ref$ref` is there we would have to string-split the content

Example of a different field:
```
r$> d$get_field("Config/Needs/verdepcheck")
[1] "rstudio/shiny, insightsengineering/teal.data, insightsengineering/teal.slice, mllg/checkmate, jeroen/jsonlite, r-lib/lifecycle, daroczig/logger, shikokuchuo/mirai, r-lib/crayon, shikokuchuo/nanonext, rstudio/renv, r-lib/rlang, daattali/shinyjs, insightsengineering/teal.code, insightsengineering/teal.logger, insightsengineering/teal.reporter, insightsengineering/teal.widgets, rstudio/bslib, yihui/knitr, bioc::MultiAssayExperiment, r-lib/R6, rstudio/rmarkdown, tidyverse/rvest, rstudio/shinytest2, rstudio/shinyvalidate, r-lib/testthat, r-lib/withr, yaml=vubiostat/r-yaml"
```

```suggestion
          if (pkg_ref$ref %in% trimws(strsplit(d$get_field("Config/Needs/DepsDev", ""), ",")[[1]]))) {
```
(a similar edit is required also below)

---

Ok, I will check the script with a real teal DESCRIPTION and see what happens.

---

There was already some code for this situation (function `parse_input`), in the new commit I reused it whenever I could